packages/namekit-react/src/components/Identity.tsx

```diff
@@ -0,0 +1,292 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  ReactNode,
+} from "react";
+import {
+  createClient,
+  Network,
+  type SecurePrimaryNameResult,
+} from "@namehash/nameguard";
+
+interface IdentityContextType {
+  network: string;
+  address: string;
+  returnNameGuardReport: boolean;
+  loadingState: "loading" | "error" | "success";
+  error?: string;
+  identityData?: SecurePrimaryNameResult;
+  followersCount?: string;
+}
+
+const IdentityContext = createContext<IdentityContextType | null>(null);
+
+const useIdentity = () => {
+  const context = useContext(IdentityContext);
+
+  if (!context) {
+    throw new Error("useIdentity must be used within an IdentityProvider");
+  }
+
+  return context;
+};
+
+interface SubComponentProps {
+  className?: string;
+  children?: ReactNode;
+}
+
+interface RootProps {
+  address: string;
+  network?: Network;
+  className?: string;
+  children: ReactNode;
+  returnNameGuardReport?: boolean;
+}
+
+const Root = ({
+  address,
+  network = "mainnet",
+  className,
+  children,
+  returnNameGuardReport = false,
+  ...props
+}: RootProps) => {
+  const [data, setData] = useState<IdentityContextType>({
+    address,
+    network,
+    returnNameGuardReport,
+    loadingState: "loading",
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setData((prev) => ({ ...prev, loadingState: "loading" }));
+
+        const nameguard = createClient({ network });
+
+        const result = await nameguard.getSecurePrimaryName(address, {
+          returnNameGuardReport,
+        });
+
+        setData((prev) => ({
+          ...prev,
+          loadingState: "success",
+          identityData: result,
+        }));
+      } catch (err) {
+        setData((prev) => ({
+          ...prev,
+          loadingState: "error",
+          error:
+            err instanceof Error ? err.message : "An unknown error occurred",
+        }));
+      }
+    };
+
+    const fetchFollowersData = async () => {
```

Comment:
Suggest we put some special care into all the names we use for things. For example, if EFP calls this idea "stats", then suggest we use similar language in our own code.